composer/lib/dependabot/composer/update_checker/version_resolver.rb

```diff
@@ -445,6 +465,11 @@ def composer_file
             dependency_files.find { |f| f.name == "composer.json" }
         end
 
+        def path_dependency_files
+          @path_dependency_files ||=
+            dependency_files.select { |f| f.name =~ /.+composer.json\z/ }
+        end
```

Comment:
Oh looks like we already have pretty much this in the composer lockfile updater: https://github.com/dependabot/dependabot-core/blob/master/composer/lib/dependabot/composer/file_updater/lockfile_updater.rb#L505-L508

Reply:
Ah yes you're right. I'm going to punt on trying to reuse anything for now, but we should really try to extract this shared logic, not the only place where this happens